### ISSUE TITLE
gradlew test     fails

### ISSUE THREAD

**Issue by vladp** [athkalia/Just-Another-Android-App#55]
not sure if I am executing the right command, but running 

gradlew test

results in 
```
...
com.example.lint.detectors.AndroidLogDetectorTest > test_detector_triggering FAILED
    junit.framework.AssertionFailedError at AndroidLogDetectorTest.java:43

com.example.lint.detectors.AndroidLogDetectorTest > test_detector_not_triggering FAILED
    junit.framework.AssertionFailedError at AndroidLogDetectorTest.java:33

4 tests completed, 2 failed
:lint:test FAILED
```
Examining the generated result (using  build/reports/tests/test/index.html)   is not particularly helpful, at least to me.

```
junit.framework.AssertionFailedError: Found compilation problems in lint test not overriding allowCompilationErrors():
AndroidLogNotExistingTestCase.java:Error: 1: The type java.lang.Object cannot be resolved. It is indirectly referenced from required .class files
AndroidLogNotExistingTestCase.java:Error: 5: Implicit super constructor Object() is undefined. Must explicitly invoke another constructor

	at junit.framework.Assert.fail(Assert.java:57)

```

**Comment by athkalia** [comment]
Hello,

thanks for the report!

after running the command I got the following exception:

```
java.lang.IllegalStateException: Iteration already started
	at android.view.ViewTreeObserver$CopyOnWriteArray.start(ViewTreeObserver.java:1122)
	at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:940)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2055)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at org.robolectric.shadows.ShadowMessageQueue.dispatchMessage(ShadowMessageQueue.java:165)
	at org.robolectric.shadows.ShadowMessageQueue.access$100(ShadowMessageQueue.java:31)
	at org.robolectric.shadows.ShadowMessageQueue$1.run(ShadowMessageQueue.java:134)
	at org.robolectric.util.Scheduler.runOrQueueRunnable(Scheduler.java:313)
	at org.robolectric.util.Scheduler.postDelayed(Scheduler.java:152)
	at org.robolectric.util.Scheduler.postDelayed(Scheduler.java:141)
	at org.robolectric.shadows.ShadowMessageQueue.enqueueMessage(ShadowMessageQueue.java:147)
	at android.os.MessageQueue.enqueueMessage(MessageQueue.java)
	at android.os.Handler.enqueueMessage(Handler.java:631)
	at android.os.Handler.sendMessageAtTime(Handler.java:600)
	at android.os.Handler.sendMessageDelayed(Handler.java:570)
	at android.os.Handler.postDelayed(Handler.java:398)
	at org.robolectric.shadows.ShadowChoreographer.postCallbackDelayed(ShadowChoreographer.java:49)
	at android.view.Choreographer.postCallbackDelayed(Choreographer.java)
	at android.view.Choreographer.postCallback(Choreographer.java:331)
	at android.view.ViewRootImpl.scheduleTraversals(ViewRootImpl.java:1079)
	at android.view.ViewRootImpl.requestLayout(ViewRootImpl.java:909)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.support.v7.widget.RecyclerView.requestLayout(RecyclerView.java:3852)
	at android.view.View.requestLayout(View.java:18728)
	at org.robolectric.internal.bytecode.ProxyMaker$GeneratedProxy/498470871.requestLayout(Unknown Source)
	at org.robolectric.shadows.ShadowView.requestLayout(ShadowView.java:234)
	at android.view.View.requestLayout(View.java)
	at android.widget.ImageView.setImageDrawable(ImageView.java:477)
	at com.bumptech.glide.request.target.GlideDrawableImageViewTarget.setResource(GlideDrawableImageViewTarget.java:80)
	at com.bumptech.glide.request.target.GlideDrawableImageViewTarget.setResource(GlideDrawableImageViewTarget.java:12)
	at com.bumptech.glide.request.target.ImageViewTarget.onResourceReady(ImageViewTarget.java:76)
	at com.bumptech.glide.request.target.GlideDrawableImageViewTarget.onResourceReady(GlideDrawableImageViewTarget.java:66)
	at com.bumptech.glide.request.target.GlideDrawableImageViewTarget.onResourceReady(GlideDrawableImageViewTarget.java:12)
	at com.bumptech.glide.request.GenericRequest.onResourceReady(GenericRequest.java:525)
	at com.bumptech.glide.request.GenericRequest.onResourceReady(GenericRequest.java:507)
	at com.bumptech.glide.load.engine.Engine.load(Engine.java:164)
	at com.bumptech.glide.request.GenericRequest.onSizeReady(GenericRequest.java:449)
	at com.bumptech.glide.request.target.ViewTarget$SizeDeterminer.notifyCbs(ViewTarget.java:177)
	at com.bumptech.glide.request.target.ViewTarget$SizeDeterminer.checkCurrentDimens(ViewTarget.java:193)
	at com.bumptech.glide.request.target.ViewTarget$SizeDeterminer.access$000(ViewTarget.java:161)
	at com.bumptech.glide.request.target.ViewTarget$SizeDeterminer$SizeDeterminerLayoutListener.onPreDraw(ViewTarget.java:292)
	at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:944)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2055)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at org.robolectric.shadows.ShadowMessageQueue.dispatchMessage(ShadowMessageQueue.java:165)
	at org.robolectric.shadows.ShadowMessageQueue.access$100(ShadowMessageQueue.java:31)
	at org.robolectric.shadows.ShadowMessageQueue$1.run(ShadowMessageQueue.java:134)
	at org.robolectric.util.Scheduler$ScheduledRunnable.run(Scheduler.java:355)
	at org.robolectric.util.Scheduler.runOneTask(Scheduler.java:257)
	at org.robolectric.util.Scheduler.advanceTo(Scheduler.java:238)
	at org.robolectric.util.Scheduler.advanceBy(Scheduler.java:221)
	at org.robolectric.util.Scheduler.advanceBy(Scheduler.java:211)
	at org.robolectric.util.Scheduler.runOrQueueRunnable(Scheduler.java:328)
	at org.robolectric.util.Scheduler.postDelayed(Scheduler.java:152)
	at org.robolectric.util.Scheduler.postDelayed(Scheduler.java:141)
	at org.robolectric.shadows.ShadowMessageQueue.enqueueMessage(ShadowMessageQueue.java:147)
	at android.os.MessageQueue.enqueueMessage(MessageQueue.java)
	at android.os.Handler.enqueueMessage(Handler.java:631)
	at android.os.Handler.sendMessageAtTime(Handler.java:600)
	at android.os.Handler.sendMessageDelayed(Handler.java:570)
	at android.os.Handler.sendMessage(Handler.java:507)
	at android.view.ViewRootImpl.dispatchResized(ViewRootImpl.java:5713)
	at org.robolectric.util.ReflectionHelpers.callInstanceMethod(ReflectionHelpers.java:231)
	at org.robolectric.android.runtime.Api23RuntimeAdapter.callViewRootImplDispatchResized(Api23RuntimeAdapter.java:47)
	at org.robolectric.android.controller.ActivityController.visible(ActivityController.java:208)
	at org.robolectric.android.controller.ActivityController.setup(ActivityController.java:245)
	at org.robolectric.Robolectric.setupActivity(Robolectric.java:97)
	at com.example.features.dashboard.view.MainActivityTest.setUp(MainActivityTest.java:34)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.robolectric.internal.SandboxTestRunner$2.evaluate(SandboxTestRunner.java:209)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:109)
	at org.robolectric.internal.SandboxTestRunner.runChild(SandboxTestRunner.java:36)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.robolectric.internal.SandboxTestRunner$1.evaluate(SandboxTestRunner.java:63)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.runTestClass(JUnitTestClassExecuter.java:114)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecuter.execute(JUnitTestClassExecuter.java:57)
	at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassProcessor.processTestClass(JUnitTestClassProcessor.java:66)
	at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:32)
	at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
	at com.sun.proxy.$Proxy3.processTestClass(Unknown Source)
	at org.gradle.api.internal.tasks.testing.worker.TestWorker.processTestClass(TestWorker.java:109)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:35)
	at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	at org.gradle.internal.remote.internal.hub.MessageHub$Handler.run(MessageHub.java:377)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
```

This has been happening for a while now in the robolectric tests and seems to be random, that's why I temporarily ignored some of the tests (seems like I should have ignored a few more). It's in my todo list to fix asap.

**Comment by athkalia** [comment]
Fixed under #59
